### PR TITLE
fix: Revert use of viper on step_create_task and step_create_meta_pipeline

### PR DESCRIPTION
**Pull request by abayer**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Description

For some reason I can't figure out, the viper addition here resulted in `--no-apply` and `--output=whatever` just not registering _at all_. So let's get rid of viper here until we can figure out what's happening.

#### Special notes for the reviewer(s)

/assign @hferentschik 

#### Which issue this PR fixes

n/a

### CODE REVIEW

**Comment by hferentschik** [comment]
> For some reason I can't figure out, the viper addition here resulted in --no-apply and --output=whatever just not registering at all. So let's get rid of viper here until we can figure out what's happening.

So you are saying that if you specify the options via the CLI they don't apply? I'll take a look. I'd rather fix the issue here. Being able to control this via env variables helps when integrating with this code.

**Comment by abayer** [comment]
@hferentschik - exactly. It seems to be some interaction with viper usage elsewhere in jx, but I'm not sure where. I'm all for getting it working, but it may make sense to merge this for now.

File: pkg/cmd/controller/pipelinerunner.go [hunk 1]
```diff
@@ -59,6 +59,8 @@ type PipelineRunnerOptions struct {
 	UseMetaPipeline      bool
 	MetaPipelineImage    string
 	SemanticRelease      bool
+	noApply              bool
```
Comment (hferentschik) [inline pkg/cmd/controller/pipelinerunner.go]: For the pipelinerunner these options really don't make much sense. They are relevant for the transitive command which is called under the hood. Hence, the env variable approach is better.

**Comment by hferentschik** [comment]
> exactly. It seems to be some interaction with viper usage elsewhere in jx, but I'm not sure where. I'm all for getting it working, but it may make sense to merge this for now.

I actually know where there rub is. Let me try something ...

**Comment by hferentschik** [comment]
I suggest something like this - https://github.com/jenkins-x/jx/pull/4628. This way we are moving forward instead of backwards.

**Comment by hferentschik** [comment]
Superseded by #4628.

**Pull request closed without merging**